arch/common/multilevel_irq.c

```diff
@@ -54,12 +60,16 @@ const struct _irq_parent_entry _lvl2_irq_list[CONFIG_NUM_2ND_LEVEL_AGGREGATORS]
 	= { LISTIFY(CONFIG_NUM_2ND_LEVEL_AGGREGATORS, CAT_2ND_LVL_LIST, (,),
 		CONFIG_2ND_LVL_ISR_TBL_OFFSET) };
 
+#ifdef CONFIG_3RD_LEVEL_INTERRUPTS
+
+BUILD_ASSERT((CONFIG_NUM_3RD_LEVEL_AGGREGATORS * CONFIG_MAX_IRQ_PER_AGGREGATOR) <=
+		     BIT(CONFIG_3RD_LEVEL_INTERRUPT_BITS),
+	     "L3 bits not enough to cover the number of L3 IRQs");
```

Comment:
I think both of these should be in units of "per aggregator". I.e. that it's incorrect to multiply the LHS side of this relationship with `CONFIG_NUM_2ND_LEVEL_AGGREGATORS`.
```suggestion
#ifdef CONFIG_3RD_LEVEL_INTERRUPTS

BUILD_ASSERT(CONFIG_MAX_IRQ_PER_AGGREGATOR <= BIT(CONFIG_3RD_LEVEL_INTERRUPT_BITS),
	     "L3 bits not enough to cover the number of L3 IRQs");
```